### DISCUSSION TITLE
Fuse Web/XML, Web/XPath, Web/EXSLT, Web/XSLT under Web/XML

### DISCUSSION THREAD

**Discussion by teoli2003** [mdn/mdn-community#357] [Content]
These 4 areas are all about XML manipulations; we should regroup them in one hierarchy and clean them a bit.

**XPath** is an expression language designed to support the query or transformation of [XML](https://en.wikipedia.org/wiki/XML) documents.
**XSLT** is a language designed to transform XML documents into other XML documents or formats, such as HTML.
**EXSLT** is a set of extensions to _XSLT_.
**XML** is just a few guides…

Note: there is already an outdated navigation bar in some of these technologies on MDN to link between them:
![Capture d’écran 2023-04-06 à 14 57 11](https://user-images.githubusercontent.com/1466293/230385420-12037f5e-7a41-4af0-82e0-899ca63efb68.png)

Regrouping them will allow:
- to add **`page-type`** to these areas
- to create/unify a **unique sidebar** for all these technologies, following the modern structure
- clean the structure of this area (that is old and sometimes broken, but is not that bad wrt the content itself)

Any thought?

If we go forward, I will create an issue with the detailed envisioned structure.

**Comment by dontcallmedom** [comment]
I think re-grouping them under a single hierarchy makes sense (although I'm not sure doing is a pre-requisite to the listed improvements)

Overall, I think the content should make it clearer that this is about XSLT and XPath as supported in browsers (since both XSLT and XPath have in the meantime evolved way beyond the browser-supported scope).

Should Web/EXSLT be kept around at all? this seems to be specifically about Firefox-supported extensions to XSLT

**Reply by teoli2003** [reply]
To be honest, I have not dug into EXSLT.

Yes, we should make it clearer that this is about XSLT/XPath in browsers.

This will need some more study before we actually start the work.

**Comment by teoli2003** [comment]
I made an inventory: https://docs.google.com/spreadsheets/d/1nwJeXKpyg2lRaQ7dbvh1vnrcCqqxSsg6yTCOlqUwPxA/edit#gid=0

It looks like we should, in addition to the fusion, do:
- a few articles should be deleted: they are landing-page with no subpages or subpages not at the same place, or outdated data
- a few articles should be merged into one: they really are paragraphs of larger articles
- the hierarchy of xpath/axes should be moved to another place.

Of course, we should also: review the articles to make them browser-agnostic (eventually renaming them) and mark what is non-standard with a banner.

For page-types, we should create the following specific types:
- xslt-element
- xpath-function
- xpath-axe
- exslt-function (likely non-standard, so we'll see if it still works somewhere)

**Reply by OnkarRuikar** [reply]
- Please merge https://github.com/mdn/yari/pull/8493 before any major restructuring.

**Reply by teoli2003** [reply]
Yes! I moved a page this morning and had to fix 30 pages manually (ok, semi-manually) afterward. This PR is so important to keep our flaws at bay.

**Comment by Rumyra** [comment]
I really like this idea @teoli2003
